examples_tests/20.Megatexture/main.cpp

```diff
@@ -435,18 +456,28 @@ int main()
         vt->commit(cm.addr, cm.texture.get(), cm.subresource, cm.uwrap, cm.vwrap, cm.border);
     }
 
-    auto gpuvt = core::make_smart_refctd_ptr<video::IGPUVirtualTexture>(driver, vt.get());
+    auto gpuvt = core::make_smart_refctd_ptr<video::IGPUVirtualTexture>(logicalDevice.get(), queue, vt.get());
 
-    auto gpuds0layout = driver->getGPUObjectsFromAssets(&ds0layout.get(), &ds0layout.get()+1)->front();
-    auto gpuds0 = driver->createGPUDescriptorSet(core::smart_refctd_ptr(gpuds0layout));//intentionally not moving layout
+    core::smart_refctd_ptr<video::IGPUDescriptorSetLayout> gpuds0layout;
+    {
+        auto gpu_array = cpu2gpu.getGPUObjectsFromAssets(&ds0layout.get(), &ds0layout.get() + 1, cpu2gpuParams);
+        if (!gpu_array || gpu_array->size() < 1u || !(*gpu_array)[0])
+            assert(false);
+
+        gpuds0layout = (*gpu_array)[0];
+    }
+
+    auto descriptorPoolDs0 = createDescriptorPool(1u); // TODO check it out
```

Comment:
not sure about this one: TODO

Reply:
you can allocate all descriptor sets from one pool if you like